### PR TITLE
feat(axe-core-4.0.1): Merge feature branch into master

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Description of changes
Update axe-core to version 4.0.1, including the following changes:
1. Update the version (using **yarn upgrade axe-core@4.0.1**
3. Disable the link-name rule due to a false positive issue (https://github.com/dequelabs/axe-core/issues/2459)
3. Update the axe-core version for the reporting package
4. Update appropriate test snapshots

We'll complete a mini-validation from the branch before merging this PR

#### Pull request checklist

- [x] Addresses an existing issue: [1759945](https://mseng.visualstudio.com/1ES/_queries/edit/1759945)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above (limited to displayed axe-core version)
- [x] (UI changes only) Verified usability with NVDA/JAWS (limited to displayed axe-core version)

### CODE REVIEW

File: src/scanner/rule-to-links-mappings.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const ruleToLinkConfiguration: DictionaryStringTo<HyperlinkDefinition[]>
     'image-redundant-alt': [BestPractice],
     'input-image-alt': [link.WCAG_1_1_1],
     'object-alt': [link.WCAG_1_1_1],
-    'link-name': [link.WCAG_2_4_4, link.WCAG_4_1_2],
+    'link-name': [/*link.WCAG_2_4_4, link.WCAG_4_1_2*/ BestPractice],
```
Comment (karanbirsingh) [inline src/scanner/rule-to-links-mappings.ts:32]: can we link to issue axe core 2459 in the code here?

Comment (DaveTryon) [reply src/scanner/rule-to-links-mappings.ts]: Added

File: src/scanner/rule-to-links-mappings.ts [hunk 2]
```diff
@@ -24,26 +24,22 @@ export const ruleToLinkConfiguration: DictionaryStringTo<HyperlinkDefinition[]>
     'landmark-unique': [BestPractice],
     'role-img-alt': [link.WCAG_1_1_1],
     'scrollable-region-focusable': [link.WCAG_2_1_1, BestPractice],
-    'area-alt': [link.WCAG_1_1_1],
+    'area-alt': [link.WCAG_1_1_1, link.WCAG_2_4_4, link.WCAG_4_1_2],
```
Comment (DaveTryon) [inline src/scanner/rule-to-links-mappings.ts:27]: Additional WCAG mapping, as called out by spec

File: src/scanner/rule-to-links-mappings.ts [hunk 3]
```diff
@@ -24,26 +24,22 @@ export const ruleToLinkConfiguration: DictionaryStringTo<HyperlinkDefinition[]>
...
     'image-alt': [link.WCAG_1_1_1],
     'image-redundant-alt': [BestPractice],
     'input-image-alt': [link.WCAG_1_1_1],
     'object-alt': [link.WCAG_1_1_1],
-    'link-name': [link.WCAG_2_4_4, link.WCAG_4_1_2],
+    'link-name': [/*link.WCAG_2_4_4, link.WCAG_4_1_2*/ BestPractice], // Temporary workaround for issue https://github.com/dequelabs/axe-core/issues/2459
     'audio-caption': [link.WCAG_1_2_1],
     'video-caption': [link.WCAG_1_2_2],
-    'video-description': [link.WCAG_1_2_5],
```
Comment (DaveTryon) [inline src/scanner/rule-to-links-mappings.ts:35]: 5 rules were removed, per spec:
- aria-dpub-role-fallback 
- checkboxgroup 
- layout-table 
- radiogroup 
- video-description

File: src/scanner/rule-to-links-mappings.ts [hunk 4]
```diff
@@ -67,7 +63,7 @@ export const ruleToLinkConfiguration: DictionaryStringTo<HyperlinkDefinition[]>
     'html-has-lang': [link.WCAG_3_1_1],
     'html-lang-valid': [link.WCAG_3_1_1],
     'valid-lang': [link.WCAG_3_1_2],
-    label: [link.WCAG_1_3_1, link.WCAG_3_3_2],
+    label: [link.WCAG_1_3_1, link.WCAG_4_1_2],
```
Comment (DaveTryon) [inline src/scanner/rule-to-links-mappings.ts:66]: From spec: label rule should be mapped to WCAG 4.1.2 instead of WCAG 3.3.2

**Pull request merged**